src/pages/workspace/WorkspaceInitialPage.tsx

```diff
@@ -99,6 +104,17 @@ function WorkspaceInitialPage({policyDraft, policy: policyProp, route}: Workspac
     const {isOffline} = useNetwork();
     const wasRendered = useRef(false);
 
+    const currentUserPolicyExpenseChat = useMemo(() => {
+        return Object.values(reports ?? {}).find((report) => {
+            // If the report has been deleted, then skip it
+            if (!report) {
+                return false;
+            }
+
+            return report.policyID === policy?.id && isPolicyExpenseChat(report) && report.ownerAccountID === currentUserPersonalDetails.accountID;
+        });
+    }, [reports, policy?.id, currentUserPersonalDetails.accountID]);
```

Comment:
Instead of getting the report here, we should just get the report id, then use `useOnyx` to subscribe to that report

Reply:
@s77rt To get the report ID, we still need to filter all of reports.

Reply:
That's okay but let's use a util function to get the id. We don't want to subscribe to the whole reports collection